jl4-core/src/L4/Utils/Ratio.hs

```diff
@@ -0,0 +1,17 @@
+module L4.Utils.Ratio (Rational, denominator, numerator, isInteger, prettyRatio) where
+
+import Data.Ratio (denominator, numerator)
+import Base
+import qualified Base.Text as Text
+
+isInteger :: Rational -> Maybe Integer
+isInteger r =
+  if denominator r == 1
+    then Just $ numerator r
+    else Nothing
```

Comment:
it's kinda interesting, that this isn't in `base`

Reply:
Yeah, I was also very surprised